pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -20,14 +21,25 @@ import (
 	"github.com/rancher/types/config"
 	"github.com/rancher/types/user"
 	"github.com/sirupsen/logrus"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
+	k8stypes "k8s.io/apimachinery/pkg/types"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 )
 
 const AgentForceDeployAnn = "io.cattle.agent.force.deploy"
 
+var (
+	nodeAgentImages    map[k8stypes.UID]string
+	nodeMutex          sync.RWMutex
```

Comment:
The Read part of the lock isn't being used, if not needed update to a standard Mutex